scripts/data/germanquad/process_data.py

```diff
@@ -0,0 +1,65 @@
+"""See clarin-knext/arguana-pl, clarin-knext/arguana-pl-qrels and
+beir.datasets.data_loader_hf.HFDataLoader for BEIR format."""
+import os
+from datasets import load_dataset, Dataset, DatasetDict, Features, Value
+
+
+dataset = load_dataset("deepset/germanquad")
+dataset.pop("train")
+
+# Deduplicate contexts and map them uniquely one-to-one to ids
+context_to_id = {}
+
+corpus_data = {"_id": [], "text": []}
+queries_data = {"_id": [], "text": []}
+qrels_data = {"query-id": [], "corpus-id": [], "score": []}
+
+for item in dataset["test"]:
+    # Check if the context is already in the dictionary
+    if item["context"] not in context_to_id:
+        context_to_id[item["context"]] = "c" + str(item["id"])
+        entry = {
+            "_id": context_to_id[item["context"]],
+            "text": item["context"]
+        }
+        corpus_data["_id"].append(entry["_id"])
+        corpus_data["text"].append(entry["text"])
+
+for item in dataset["test"]:
+    entry = {
+        "_id": "q" + str(item["id"]),
+        "text": item["question"]
+    }
+    queries_data["_id"].append(entry["_id"])
+    queries_data["text"].append(entry["text"])
```

Comment:
This could be refactored into the loop above, since I ditched writing to JSONL...
I find the current way a bit more comprehendible though